src/main/java/com/google/api/codegen/config/ProtoField.java

```diff
@@ -61,6 +63,16 @@ public String getFullName() {
   }
 
   @Override
+  public Name getNameAsParameterName() {
+    return Name.from(getSimpleName());
+  }
+
+  @Override
+  public String getNameAsParameter() {
+    return getNameAsParameterName().toLowerUnderscore();
```

Comment:
token formatting (camel/underscore) should be up to the target language

Reply:
Should I make a "gapic config" SurfaceNamer? This is used in config-gen.

Reply:
Or should I assume that the gapic configs are case-insensitive, and adjust the transformers accordingly?

Reply:
Oh, well "gapic config" is only one language, so actually I think what you have is fine.